File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
public class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        List<Integer> answerList = new ArrayList<>();
        ArrayList<int[]> list = new ArrayList<>();
        for(int i =0; i < priorities.length; i++){ //0번쨰 배열은 priorities의 값, 1번째 배열은 순서
            list.add(new int[]{priorities[i],i});
        }
        while(list.size() != 0){
            int[] temp = list.get(0);
            boolean isPrior = true;
            for (int i = 0; i < list.size(); i++) {
                if (list.get(i)[0]>temp[0]) {
                    list.add(temp);
                    list.remove(0);
                    isPrior = false;
                    break;
                }
            }
            if (isPrior){
                answerList.add(temp[1]);
                list.remove(0);
            }
        }
        for (int i = 0; i < answerList.size(); i++) {
            if (answerList.get(i)==location){
                answer = i+1;
                break;
            }
        }
        return answer;
    }
}

File: sejeong/프로세스.py
def solution(priorities,location):
    q = []
    answer = 0
    for idx,pri in enumerate(priorities):
        q.append((idx,pri))  # q에다가 인덱스와 우선순위를 같이 넣기
    while q:
        max_v = max(q,key=lambda x:x[1])[1]  # 우선순위가 가장높은 것 찾기
        t =q.pop(0)
        if t[1] < max_v: # 우선순위가 안높으면 q의 뒤로 보내기
            q.append(t)
        else:   # 가장높은 우선순위이면 실행하므로 실행횟수세기
            answer +=1
            if t[0] == location: # 우리가 찾는 인덱스위치이면
                return answer   # 실행횟수 출력



priorities = [1, 1, 9, 1, 1, 1]
location = 0
print(solution(priorities,location))